vehicle/OVMS.V3/components/vehicle_maxus_edeliver3/src/vehicle_med3.cpp

```diff
@@ -29,117 +31,176 @@
-        int value1 = (int)data[0];
-        int value2 = ((int)data[0] << 8) + (int)data[1];
-                
-        switch (pid)
-            {
-                case 0xf190:  // VIN
-                    strncat(m_vin,(char*)data,length);
-                    if (mlremain==0)
-                    {
-                        StandardMetrics.ms_v_vin->SetValue(m_vin);
-                        m_vin[0] = 0;
-                    }
-                    break;
-                case 0xe001: //soh
-                    StandardMetrics.ms_v_bat_soh->SetValue(value1);
-                    break;
-                case 0xe002: //soc scaled from 2 - 99
-                    StandardMetrics.ms_v_bat_soc->SetValue(value1);
-                    break;
-                case 0xe005:  // temperature??
-                    StandardMetrics.ms_v_mot_temp->SetValue(value1 / 10.0f);
-                    break;
-                case 0xe006:  // temperature??
-                    StandardMetrics.ms_v_env_temp->SetValue(value1 / 10.0f);
-                    break;
-                case 0xe019:  // Pack Voltage
-                    StandardMetrics.ms_v_bat_voltage->SetValue(value2 / 10.0f);
-                    break;
-                case 0xe022:
-                    StandardMetrics.ms_v_bat_12v_current->SetValue(value1 / 10.0f);
-                    break;
-                case 0xe036:
-                    StandardMetrics.ms_v_charge_voltage->SetValue(value1); // possible but always 224 untill found only
-                    break;
-                case 0xe039:
-                    StandardMetrics.ms_v_charge_current->SetValue(value1);
-                    StandardMetrics.ms_v_charge_climit->SetValue(value1);
-                    StandardMetrics.ms_v_charge_power->SetValue((StandardMetrics.ms_v_charge_current->AsFloat() * (StandardMetrics.ms_v_charge_voltage->AsFloat() )) / 1000); // work out current untill pid found * (value / 10.0f) / 1000.0f
-                    break;
-                
-            }
+{
+  // init / fill rx buffer:
+  if (m_poll_ml_frame == 0) {
+    m_rxbuf.clear();
+    m_rxbuf.reserve(length + mlremain);
+  }
+  m_rxbuf.append((char*)data, length);
+  if (mlremain)
+    return;
+
+  // response complete:
+    ESP_LOGV(TAG, "IncomingPollReply: PID %02X: len=%d %s", pid, m_rxbuf.size(), hexencode(m_rxbuf).c_str());
     
-        // init / fill rx buffer:
-        if (m_poll_ml_frame == 0)
-            {
-                rxbuf.clear();
-                rxbuf.reserve(length + mlremain);
-            }
-                rxbuf.append((char*)data, length);
-        if (mlremain)
-        {}
+    int value1 = (int)data[0];
+    int value2 = ((int)data[0] << 8) + (int)data[1];
+    
+  switch (pid)
+  {
+      case vcuvin:  // VIN
+          strncat(m_vin,(char*)data,length);
+          if (mlremain==0)
+          {
+              StandardMetrics.ms_v_vin->SetValue(m_vin);
+              m_vin[0] = 0;
+          }
+          break;
```

Comment:
You now have an assembled VIN in m_rxbuf, the assembly by strncat() isn't needed and won't work any longer. You only need to do `StandardMetrics.ms_v_vin->SetValue(m_rxbuf);` here. You also no longer need the `m_vin` buffer.